core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -46,7 +46,7 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
   /** Create a SparkConf that loads defaults from system properties and the classpath */
   def this() = this(true)
 
-  private[spark] val settings = new HashMap[String, String]()
+  private[spark] val settings = new HashMap[String, String] with SynchronizedMap[String, String]
```

Comment:
Then which one do you suggest?

We also hit a bug related to java.util.ConcurrentMap recently. Should we use TrieMap ? Actually I tried it at first, but not sure what's quality it has, the follow the docs here:  http://docs.scala-lang.org/overviews/collections/maps.html